### PR TITLE
Add documentation for gdal based readers.

### PR DESCRIPTION
**Pull request by milos-colic**

### CODE REVIEW

File: docs/source/api/vector-format-readers.rst [hunk 1]
```diff
@@ -0,0 +1,243 @@
+=====================
+Vector Format Readers
+=====================
+
+
+Intro
+################
+Mosaic provides spark readers for vector files supported by GDAL OGR drivers.
+Only the drivers that are built by default are supported.
```
Comment (mjohns-databricks) [inline docs/source/api/vector-format-readers.rst:9]: @milos-colic should you enumerate a list of supported vector files here (as was done for raster?)

Comment (milos-colic) [reply docs/source/api/vector-format-readers.rst:9]: In raster we do filter out to a subset of GDAL default drivers.
In vector we dont filter, so I just provided the reference to GDAL OGR drivers.

Comment (milos-colic) [reply docs/source/api/vector-format-readers.rst:9]: That is a valid point, I will add some of them and update the PR.

Comment (mjohns-databricks) [reply docs/source/api/vector-format-readers.rst:9]: Here are some common useful file formats with driver short name hyperlinked, For more information please refer to gdal vector driver [documentation](https://gdal.org/drivers/vector/index.html): 

- [GeoJSON](https://gdal.org/drivers/vector/geojson.html#vector-geojson) (also [ESRIJSON](https://gdal.org/drivers/vector/esrijson.html#vector-esrijson), [TopoJSON](https://gdal.org/drivers/vector/topojson.html#vector-topojson))
- ESRI File Geodatabase ([FileGDB](https://gdal.org/drivers/vector/filegdb.html#vector-filegdb)) and ESRI File Geodatabase vector ([OpenFileGDB](https://gdal.org/drivers/vector/openfilegdb.html#vector-openfilegdb)) - _Mosaic implements named reader `geo_db` (described in this doc)_
- ESRI Shapefile / DBF ([ESRI Shapefile](https://gdal.org/drivers/vector/shapefile.html#vector-shapefile)) - _Mosaic implements named reader `shapefile` (described in this doc)_
- Network Common Data Form ([netCDF](https://gdal.org/drivers/vector/netcdf.html#vector-netcdf)) - _Mosaic implements raster reader also_
- (Geo)Parquet ([Parquet](https://gdal.org/drivers/vector/parquet.html#vector-parquet)) - _Mosaic will be implementing a custom reader soon_
- Spreadsheets ([XLSX](https://gdal.org/drivers/vector/xlsx.html#vector-xlsx), [XLS](https://gdal.org/drivers/vector/xls.html#vector-xls), [ODS](https://gdal.org/drivers/vector/ods.html#vector-ods))
- U.S. Census TIGER/Line ([TIGER](https://gdal.org/drivers/vector/tiger.html#vector-tiger))
- PostgreSQL Dump ([PGDump](https://gdal.org/drivers/vector/pgdump.html#vector-pgdump))
- Keyhole Markup Language ([KML](https://gdal.org/drivers/vector/kml.html#vector-kml))
- Geography Markup Language ([GML](https://gdal.org/drivers/vector/gml.html#vector-gml))
- [GRASS](https://gdal.org/drivers/vector/grass.html#vector-grass) - option for Linear Referencing Systems ([LRS](https://grass.osgeo.org/grass82/manuals/lrs.html))

**Review by mjohns-databricks — APPROVED** [review]
LGTM

**Pull request merged**